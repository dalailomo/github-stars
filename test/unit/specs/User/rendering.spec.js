import Vue from 'vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import User from '@/components/User'
import { getWrapper } from './../../helpers'
import { emptyState, stateWithUser, mocks } from './mocks'

Vue.use(Vuex)
Vue.use(Vuetify)

describe('User rendering', () => {
  const actions = {
    actionGetGithubUser: jest.fn(),
  }

  it('Renders that there is no user', () => {
    const wrapper = getWrapper(User, actions, emptyState, mocks, {})

    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })

  it('Renders user details', () => {
    const wrapper = getWrapper(User, actions, stateWithUser, mocks, {})

    const template = wrapper.html()

    expect(template).toMatchSnapshot()
  })
})
