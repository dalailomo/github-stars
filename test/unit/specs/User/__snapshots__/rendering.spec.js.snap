// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User rendering Renders that there is no user 1`] = `
<div class="profile card" style="height: auto;">
  User not found :(
</div>
`;

exports[`User rendering Renders user details 1`] = `
<div class="profile card" style="height: auto;">
  <div class="white--text card__media" style="height: 200px;">
    <div class="card__media__background"></div>
    <div class="card__media__content">
      <div class="container fill-height fluid">
        <div class="layout fill-height">
          <div class="flex xs12 align-end flexbox"><span class="headline">foo</span></div>
        </div>
      </div>
    </div>
  </div>
  <div class="card__title">
    <div><span class="grey--text">foo</span>
      <br> <span>Foo</span></div>
  </div>
  <div class="card__actions">
    <div class="spacer"></div>
    <button type="button" class="btn btn--flat btn--small primary--text">
      <div class="btn__content">View user page</div>
    </button>
  </div>
</div>
`;
